functions/01_TotalActivation/Temporal_PFM/PFM_Temporal_OneTimeCourse.m

```diff
@@ -11,14 +11,14 @@
 % - ParametersOut: structure containing the parameters of the algorithm
 %
 % Implemented by Eneko Uruñuela, 13.12.2022
-function [x,ParametersOut] = TA_Temporal_OneTimeCourse(y,idx_vox,ParametersIn)
+function [Activity_related,Activity_inducing,Innovation,ParametersOut] = TA_Temporal_OneTimeCourse(y,idx_vox,ParametersIn)
 
     % The necessary HRF matrices are computed
-    X_tilde = param.HRF;
-    X_tilde_trans = param.X_tilde_trans;
-    X_tilde_tt = param.X_tilde_tt;
-    c_ist = param.c_ist
-
+    ParametersOut = ParametersIn;
```

Comment:
The param structure has to maintain all its fields during iterations and you weren't copyng them